vernac/classes.ml

```diff
@@ -195,10 +195,9 @@ let new_instance ?(abstract=false) ?(global=false) ?(refine= !refine_instance)
 	in
 	let (_, ty_constr) = instance_constructor (k,u) subst in
         let termtype = it_mkProd_or_LetIn ty_constr (ctx' @ ctx) in
-        let sigma = Evd.minimize_universes sigma in
         Pretyping.check_evars env Evd.empty sigma termtype;
+        let sigma, termtype = Evarutil.finalize env sigma (fun nf -> nf termtype) in
```

Comment:
Here we restrict where we didn't use to (this is Declare Instance which I think is quite rare).